### ISSUE TITLE
PostgreSQL search `get_weight()` function accesses `weight` variable before assignment

### ISSUE THREAD

**Issue by isolationism** [wagtail/wagtail#10289]
I'm not sure what I've done to reveal this otherwise uncaught long-standing defect, but there is a simple programming error stopping me cold from updating any custom image model data:

https://github.com/wagtail/wagtail/blob/b504403ebf4e0b675f1be26ff83f2923c5947a98/wagtail/search/backends/database/postgres/weights.py#L59

The problem is per the title: `local variable 'weight' referenced before assignment`. I don't know why nobody else is getting the error. My `boost` argument value is 10.

**Comment by isolationism** [comment]
Helpful additional detail:

Defect was revealed by saving model update *without* the `wagtail.search` app listed in the `INSTALLED_APPS` in Django. After adding it and running migrations, updates succeed normally.

**Comment by gasman** [comment]
Thanks for the report @isolationism!

As you note, this will only occur when `wagtail.search` is missing from INSTALLED_APPS. If `wagtail.search` is present, then [`set_weights` will be called on startup](https://github.com/wagtail/wagtail/blob/0447b259c7d65d4187b8eef8faeeddc87fee8e23/wagtail/search/apps.py#L24), and this will ensure that the `BOOSTS_WEIGHTS` list is non-empty. Consequently, if the `for max_boost, weight in BOOSTS_WEIGHTS` loop completes without returning, `weight` will still be set to the last value we saw in the list, which is exactly what we want to happen - in this case the boost value is lower than any of the values supported by the database, so we pick the lowest weight.

`wagtail.search` is a required app, and the `wagtail start` command and [documentation for integrating Wagtail into an existing project](https://docs.wagtail.org/en/stable/getting_started/integrating_into_django.html) consistently include it - so the presence of an error when omitting it is not a bug.

Having said that, I'd certainly want it to fail with a more informative error message that points towards the missing INSTALLED_APPS entry. After some investigation, though, I can't find a way to get it to fail with this error - if I leave `wagtail.search` out, things mostly work up to the point that I actually run a search - and without that it's hard to know where the extra error checking should go. I suspect that this might only happen in a very esoteric set of circumstances, in which case an esoteric error message might be unavoidable... however, if you can provide a set of steps starting from a fresh Wagtail installation that lead to this error, I'll be happy to take another look.

**Comment by indepndnt** [comment]
This issue and #8713 seem to be the same issue and both show up in my google search for the error message I got.

I see that the solution to my problem is to include `wagtail.search` in the installed apps. I'm new to Wagtail and I don't know of a better place to document what I've learned here (and will surely forget by tomorrow), so my apologies in advance if commenting on a closed issue is frowned upon.

Producing the error: I got this error by creating a Wagtail project, removing `wagtail.search` from `INSTALLED_APPS`, creating a `Page` model with a foreign key to `wagtailimages.Image`, then trying to upload an image to that field in admin.

Documentation: The "Integrating Wagtail into a Django project" in the "Getting started" section does indeed say "add the following apps" and lists `wagtail.search`. However, on the "How to add Wagtail into an existing Django project" page in the "Advanced topics" section under "Apps", it says "Wagtail was built to be modular, so many Wagtail apps can be omitted to suit your needs." It does go on to list what the apps are for, but I was unable to find any indication of which apps are required and which could safely be omitted.

This is the full model for which uploading an image triggered the error.
```
class IndexPage(Page):
    lead = RichTextField(blank=True, help_text="Lead text")
    feature_image = models.ForeignKey(
        "wagtailimages.Image",
        null=True,
        blank=True,
        on_delete=models.SET_NULL,
        related_name="+",
        help_text="Featured product image",
    )
    feature_text = RichTextField(blank=True, help_text="Featured product description")
    body = RichTextField(blank=True)

    content_panels = Page.content_panels + [
        FieldPanel("lead"),
        MultiFieldPanel([FieldPanel("feature_image"), FieldPanel("feature_text")], heading="Featured product"),
        FieldPanel("body"),
    ]
```
